src/transaction.rs

```diff
@@ -195,7 +197,7 @@ impl<'conn> Transaction<'conn> {
             return Ok(());
         }
         match self.drop_behavior() {
-            DropBehavior::Commit => self.commit_(),
+            DropBehavior::Commit => self.commit_().or_else(|_| self.rollback_()),
```

Comment:
I am not sure but the `.or_else(|_| self.rollback_())` should only be done in `drop` (in case the user wants to wait and retry for example after `finish`).
And maybe the `rollback_` should only be called for some specific error code(s).

Reply:
It's not possible to retry if `finish()` fails, because it consumes `self` and drops it.

Reply:
Right, sorry.